test/README.md

```diff
@@ -0,0 +1,67 @@
+Candid test data
+================
+
+This directory contains Candid test data, to be used to check Candid
+implementations for compliance.
+
+This test suite covers
+
+ * parsing Candid service descriptions
+ * given a Candid type, decoding binary Candid data, asserting success or failure
+ * given a Candid type, decoding textual Candid data, asserting success or failure
+
+Because this test data necessarily has to be host-language agnostic, it is not
+possible to
+
+ * assert the decoded value, as it would be host-language specific
+ * cover encoding, as it would require specifying a host-language value,
+   and also because Candid encoding is not canonical
+
+Implementers are advised to extend this test suite with randomzied
+round-tripping tests (host value → candid → host value).
+
+Format
+======
+
+The test suite contains these kind of files:
+
+ * `<name>.good.did`:
+   A Candid service description, with valid candid syntax.
+   Implementations should be able to parse them.
+
+ * `<name>.bad.did`:
+   A Candid service description, with invalid candid syntax.
+   Implementations should reject parsing them.
+
+ * `<name>.test.did`:
+   A set of Candid tests, written in a grammar that extens the Candid type and value grammar:
+
+   The root production is
+   ```
+   <enctest>  ::= <def>;* <test>;*
+   ```
+   to allow any number of type definitions followed by encoding tests with grammar
+   ```
+   <test> = <argseq> <input> <outcome>
```

Comment:
```suggestion
   <test> = <argseq> <input> <outcome> <desc>
```

Reply:
`<argseq>` can be optional. Unless we are testing numbers or subtyping, we don't need to specify them.

Reply:
Not sure it should be optional; some host languages will only support “parsing given an expected type” (motoko, for example). The ability to parse generic binary data, without an expected type, should not be expected from all implementations.